provisioning.go

```diff
@@ -0,0 +1,370 @@
+// this function. If the wrong value is provided for either authorization, then a AuthFailError error will be returned. If the
+// correct authorization values are not known, then the only way to recover from this is to call the function with mode set to
+// ProvisionModeClear. If there is an object already stored at the location used for the endorsement key then this function will
+// evict it automatically from the TPM.
+//
+// This function will create and persist a storage root key, which requires knowledge of the authorization value for the storage
+// hierarchy. If called with mode set to ProvisionModeClear, or if called just after clearing the TPM via the physical presence
+// interface, the authorization value for the storage hierarchy will be empty at the point that it is required. If called with any
+// other mode and if the authorization value for the storage hierarchy has previously been set, it will need to be provided by calling
+// TPMConnection.OwnerHandleContext().SetAuthValue() prior to calling this function. If the wrong value is provided for the storage
+// hierarchy authorization, then a AuthFailError error will be returned. If the correct authorization value is not known, then the
+// only way to recover from this is to call the function with mode set to ProvisionModeClear. If there is an object already stored at
+// the location used for the storage root key then this function will evict it automatically from the TPM.
+//
+// This function will create a pair of NV indices used for locking access to sealed key objects, if necessary. These indices will be
+// created at 0x01801100 and 0x01801101. This requires knowledge of the authorization value for the storage hierarchy. If called with
+// mode set to ProvisionModeClear, or if called just after clearing the TPM via the physical presence interface, the authorization
+// value for the storage hierarchy will be empty at the point that it is required. If called with any other mode and if the
+// authorization value for the storage hierarchy has previously been set, it will need to be provided by calling
+// TPMConnection.OwnerHandleContext().SetAuthValue() prior to calling this function. If the wrong value is provided for the storage
+// hierarchy authorization, then a AuthFailError error will be returned. If the correct authorization value is not known and new NV
+// indices need to be created, then the only way to recover from this is to call the function with mode set to ProvisionModeClear. If
+// there are already NV indices defined at either of these handles but they don't meet the requirements of this function, a
+// TPMResourceExistsError error will be returned. In this case, the caller will either need to manually undefined these using
+// TPMContext.NVUndefineSpace, or clear the TPM.
+//
+// If mode is not ProvisionModeWithoutLockout, the authorization value for the lockout hierarchy will be set to newLockoutAuth.
```

Comment:
it is in parenthetical earlier but probably this should repeat that we will also disable owner clear.

Reply:
I ended up refactoring this bit of documentation quite a bit to try to factor in some of these comments but also reduce some of the duplication.